pkg/activator/net/revision_backends.go

```diff
@@ -220,8 +227,19 @@ func (rw *revisionWatcher) sendUpdate(clusterIP string, dests sets.String) {
 
 // checkDests performs probing and potentially sends a dests update. It is
 // assumed this method is not called concurrently.
-func (rw *revisionWatcher) checkDests(dests sets.String) {
-	if len(dests) == 0 {
+func (rw *revisionWatcher) checkDests(dests destSets) {
+	// TODO(taragu) the readyDests and notReadyDests should have different
```

Comment:
Based on this comment, should you not keep them separate so far?

Reply:
I see. I've made the change to combine them only when it comes to probing. We are still only going to send ready dests if cluster ip is healthy.